sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -170,6 +222,8 @@ object BasicWriteJobStatsTracker {
   private val NUM_OUTPUT_BYTES_KEY = "numOutputBytes"
   private val NUM_OUTPUT_ROWS_KEY = "numOutputRows"
   private val NUM_PARTS_KEY = "numParts"
+  /** XAttr key of the data length header added in HADOOP-17414. */
```

Comment:
`HADOOP-17414` seems to be not committed yet and is still targeting 3.3.1.
Shall we hold one this PR until we have an official Hadoop release?

Reply:
I'm developing them in sync so they can be done end to end. Yes, it makes sense for the Hadoop one to be committed,  first -but if reviewing the spark one finds changes which need to be done there (is the header a good name?) then an initial review of this PR will make sure that they are lined up